### PR TITLE
Remove won't share identifiable information bullet for Assistant code sharing

### PR DESCRIPTION
**Pull request by Sly1029**
# Thanks for improving Semgrep Docs 😀

Discussed with Jonathan and verified that this ok to do. This removal should go out after messaging is finished about this change

### Please ensure

- [X] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [X] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: docs/semgrep-assistant/overview.md [hunk 1]
```diff
@@ -124,7 +124,6 @@ Semgrep uses API permissions to access code on your pre-selected GitHub or GitLa
 * Semgrep, Inc. is a paying customer of OpenAI and has a Data Protection Agreement signed with them (provided upon request by [contacting support](/docs/support). The code snippets we upload are persisted by OpenAI temporarily, following their data usage policies at [Enterprise privacy at OpenAI](https://openai.com/enterprise-privacy).
 <!-- markdown-link-check-enable -->
 * Semgrep, Inc. takes the following steps to protect data that is processed by AI since Assistant requires the sharing of code snippets with a third party:
-  * Semgrep shares code snippets with OpenAI without identifying the customer or repository name.
```
Comment (s-santillan) [inline docs/semgrep-assistant/overview.md:127]: @jsmoxon What exactly has changed here - do we share code snippets with openai at all? is it possible to at least keep part of this bullet (let's say we don't identify the customer, but we do send the repository name)?

Comment (jsmoxon) [reply docs/semgrep-assistant/overview.md:127]: We do share code snippets with OpenAI. What has changed is that we now sometimes share the repository name. And sometimes it is possible to determine the customer name from the repository name. E.g. if the repo is called 'acme-foo-service`. 

Previously we didn't share the repo name. @Sly1029 correct me if I have this wrong.

Comment (Sly1029) [reply docs/semgrep-assistant/overview.md:127]: Yeah now we have some desire to start sharing this information. But I wanted to leave it up to Jack to do the comms first

Comment (Sly1029) [reply docs/semgrep-assistant/overview.md:127]: Ah yeah that second option sounds good

**Review by s-santillan — APPROVED** [review]
Approving for now, we'll make a follow up PR if there are any changes.

**Pull request merged**